ddtrace/internal/flare.py

```diff
@@ -29,13 +27,28 @@
 log = get_logger(__name__)
 
 
+class FlarePrepRequest:
+    def __init__(self, log_level: str):
+        if logging.getLevelName(log_level) == f"Level {log_level}":
+            raise TypeError("Invalid log level provided: %s", log_level)
+        self.log_level = log_level
+
+
+class FlareSendRequest:
+    def __init__(self, case_id: str, hostname: str, email: str):
+        self.case_id = case_id
+        self.source = "tracer_python"
+        self.hostname = hostname
+        self.email = email
```

Comment:
you can probably do this as an actual dataclass

```suggestion
@dataclasses.dataclass
class FlareSendRequest:
    case_id: str
    hostname: str
    email: str
    source: str = "tracer_python"
```

Same for the one above, you could use a [__post_init__](https://docs.python.org/3/library/dataclasses.html#dataclasses.__post_init__) to do the checking/validation of the level.